core/store/migrate/migrations/0254_log_poller_primary_keys.sql

```diff
@@ -0,0 +1,24 @@
+-- +goose Up
+
+ALTER TABLE evm.logs DROP CONSTRAINT logs_pkey;
+ALTER TABLE evm.logs ADD COLUMN id BIGINT PRIMARY KEY GENERATED ALWAYS AS IDENTITY;
+CREATE UNIQUE INDEX idx_logs_chain_block_logindex ON evm.logs (evm_chain_id, block_number, log_index);
```

Comment:
That's a good point, I'll check on this.

Reply:
After checking on this, I realized why I originally made it an index.

I don't think it's true about a unique constraint being more lightweight. Every unique constraint generates an unique index automatically... the only difference is we pick the name here rather than it picking one automatically.

I mentioned in another comment that we need at least (chain id, block number) as an index, for the DeleteLogsAndBlocksAfter query. But if we add a unique constraint that generated an entire second index which takes up twice the space. By doing it this way, all we have to do is add log_index to the end and then there is no need to have 2 separate indexes... which is what we had before (due to the one generated by the primary key).

In the end I decided it seems best to leave everything as-is, but I added a lot of comments explaining what's going on and separated the tables out as you suggested.